flink-filesystems/flink-s3-fs-hadoop/src/main/java/org/apache/flink/fs/s3hadoop/S3FileSystemFactory.java

```diff
@@ -132,7 +132,12 @@ else if (scheme != null && authority == null) {
 			final S3AFileSystem fs = new S3AFileSystem();
 			fs.initialize(fsUri, hadoopConfig);
 
-			return new HadoopFileSystem(fs);
+			if (flinkConfig != null) {
```

Comment:
Can you pull this if into `HadoopUtils.limitIfConfigured` method? That way those invocations would simply look like: `limitIfConfigured(new HadoopFileSystem(fs), scheme, Optional.ofNullable(flinkConfig))` without duplicated it/else branch.